### DISCUSSION TITLE
Can you do a dispatcher based on JWT contents?

### DISCUSSION THREAD

**Discussion by mcrobbj** [microcks/microcks#760] [Q&A]
So for example I have different error responses for first problem found with a JWT
- sig invalid
- aud invalid
- iss invalid
- expired
- scope invalid
- etc

**Comment by lbroudoux** [comment]
Hi @mcrobbj 

Sorry for the delay in response (was very busy these last weeks). Currently, you should be able to do this using the `SCRIPT` dispatcher, accessing the `Authentication` header and decoding/analyzing your JWT content using groovy functions. See some `SCRIPT` dispatching examples here: https://microcks.io/documentation/using/advanced/dispatching/#common-use-cases

You may also find some interesting primitives here: https://loopednetwork.medium.com/creating-a-jwt-in-the-groovy-programming-language-be11637a3f49

Hope it helps!
